### PR TITLE
TXL-216: Allow question sequence out of strict Scripture reference order.

### PR DESCRIPTION
**Pull request by tombogle**

### CODE REVIEW

File: Transcelerator/NewQuestionDlg.cs [hunk 1]
```diff
@@ -118,22 +108,11 @@ private int EndVerse
...
-				if (m_dataGridViewExistingQuestions.RowCount == 0)
-					return 1;
-				
-				return ((TranslatablePhrase)m_dataGridViewExistingQuestions.Rows[0].DataBoundItem).SequenceNumber +
-					SelectedInsertionLocation;
-			}
-		}
+		public int SequenceNumber =>
+			BasePhrase == null ? 0 :
+				BasePhrase.SequenceNumber + (InsertBeforeBasePhrase ? 0 : 1);
```
Comment (FoolRunning) [inline Transcelerator/NewQuestionDlg.cs:115]: Since this property now works off of BasePhrase, the behavior has changed since it now looks at the selected rows and also behaves differently when there are more than one. Was this change intentional?

Comment (tombogle) [reply Transcelerator/NewQuestionDlg.cs:120]: The Sequence number used to be a 1-based index for the questions in a particular verse (or verse range). Now it uniquely identifies this phrase within a given category of a particular section. I have added a comment here and updated the comment on TranslatablePhrase.SequenceNumber.

File: Transcelerator/NewQuestionDlg.cs [hunk 2]
```diff
@@ -212,21 +202,46 @@ public TranslatablePhrase BasePhrase
 			m_scrPsgReference.PassageChanged += r =>
 			{
```
[the remark this answers is not in the document]
Comment (tombogle) [reply Transcelerator/NewQuestionDlg.cs:208]: I'm guessing it was done to avoid having this event fire when the control was first initialized. I vaguely assumed this was somehow related to the comment about overcoming the HACK in ScrPassageControl (to prevent setting m_existingStartRef prematurely?), but maybe not.

File: Transcelerator/NewQuestionDlg.cs [hunk 3]
```diff
@@ -424,5 +464,18 @@ private void m_txtVernacularQuestion_Leave(object sender, EventArgs e)
...
+			if (DialogResult.OK ==
+				MessageBox.Show(this, "Sorry this feature is not available yet. But if you are " +
+				"connected to the Internet and have not opted out of transmitting analytics data, " +
+				"this feature will now be requested for you.",
+				"Transcelerator Feature Request", MessageBoxButtons.OKCancel))
+			{
+				Analytics.Track("TXL-218", new Dictionary<string, string>
+					{{"StartRef", StartReference.ToString(BCVRef.RefStringFormat.General)}});
+			}
+		}
```
[the remark this answers is not in the document]
Comment (tombogle) [reply Transcelerator/NewQuestionDlg.cs:484]: Maybe so. Glad you liked it.
